src/main/scala/fr/acinq/bitcoin/DeterministicWallet.scala

```diff
@@ -29,8 +31,8 @@ object DeterministicWallet {
       */
     def apply(path: String) : KeyPath = {
       def toNumber(value: String): Long = if (value.last == '\'') hardened(value.dropRight(1).toLong) else value.toLong
-      val path1 = path.stripPrefix("m/").stripPrefix("/")
-      new KeyPath(path1.split('/').map(toNumber))
+      val path1 = path.stripPrefix("m").stripPrefix("/")
+      if (path1.isEmpty) KeyPath.Root else KeyPath(path1.split('/').map(toNumber))
```

Comment:
it's cleaner and common practice to use `new KeyPath(...)` here

Reply:
Re-added